timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/ClientPaxosResourceFactory.java

```diff
@@ -0,0 +1,264 @@
+            PaxosLearner localLearner = components.learner(client);
+            LocalAndRemotes<PaxosLearner> allLearners =
+                    instrument(metrics, PaxosLearner.class, localLearner, remoteLearners);
+            return new SingleLeaderLearnerNetworkClient(
+                    allLearners.local(),
+                    allLearners.remotes(),
+                    quorumSize,
+                    allLearners.withSharedExecutor(sharedExecutor));
+        };
+
+        return ImmutableNetworkClientFactories.builder()
+                .acceptor(acceptorClientFactory)
+                .learner(learnerClientFactory)
+                .build();
+    }
+
+    private static NetworkClientFactories batch(
+            MetricRegistry metricRegistry,
+            TimeLockInstallConfiguration install,
+            int quorumSize,
+            ExecutorService sharedExecutor,
+            UseCaseAwareBatchPaxosResource resource) {
+        // TODO(fdesouza): pass these in, as they become use case aware
+        List<BatchPaxosAcceptorRpcClient> remoteBatchAcceptors = createProxies(
+                BatchPaxosAcceptorRpcClient.class,
+                "timestamp-bound-store.batch-acceptor",
+                metricRegistry,
+                install);
+
+        List<BatchPaxosAcceptor> allBatchAcceptors = instrument(
+                metricRegistry,
+                BatchPaxosAcceptor.class,
+                resource.acceptor(PaxosUseCase.TIMESTAMP).asLocalBatchPaxosAcceptor(),
+                UseCaseAwareBatchPaxosAcceptorAdapter.wrap(PaxosUseCase.TIMESTAMP, remoteBatchAcceptors)).all();
+
+        AutobatchingPaxosAcceptorNetworkClientFactory acceptorFactory =
+                AutobatchingPaxosAcceptorNetworkClientFactory.create(allBatchAcceptors, sharedExecutor, quorumSize);
+
+        // TODO(fdesouza): pass these in as they become use case aware
+        List<BatchPaxosLearnerRpcClient> remoteBatchLearners = createProxies(
+                BatchPaxosLearnerRpcClient.class,
+                "timestamp-bound-store.batch-learner",
+                metricRegistry,
+                install);
+
+        List<BatchPaxosLearner> allBatchLearners = instrument(
+                metricRegistry,
+                BatchPaxosLearner.class,
+                resource.learner(PaxosUseCase.TIMESTAMP),
+                UseCaseAwareBatchPaxosLearnerAdapter.wrap(PaxosUseCase.TIMESTAMP, remoteBatchLearners)).all();
+
+        AutobatchingPaxosLearnerNetworkClientFactory learnerFactory =
+                AutobatchingPaxosLearnerNetworkClientFactory.create(allBatchLearners, sharedExecutor, quorumSize);
+
+        return ImmutableNetworkClientFactories.builder()
+                .acceptor(acceptorFactory::paxosAcceptorForClient)
+                .learner(learnerFactory::paxosLearnerForClient)
+                .addCloseables(acceptorFactory, learnerFactory)
+                .build();
+    }
```

Comment:
nit: This is getting pretty long in general - I think I'd prefer if we split the `NetworkClientFactoriesFactory` (sigh) piece of this out to its own class

Reply:
this has made the PR size go up, but the additions are mainly imports and whitespace lol from having many separate files, really wish I had dagger 😣